supabase/migrations/47_registered_avro_schemas.sql

```diff
@@ -0,0 +1,32 @@
+
+create table registered_avro_schemas (
+  like internal._model including all,
+
+  avro_schema      json not null,
+  avro_schema_md5  text generated always as (md5(trim(avro_schema::text))) stored,
+  catalog_name     catalog_name not null,
+  registry_id      serial unique not null
+);
+
+create index idx_registered_avro_schemas_avro_schema_md5 on registered_avro_schemas (avro_schema_md5);
+
+comment on table registered_avro_schemas is '
+Avro schemas registered with a globally unique, stable registery ID.
+
+This is used to emulate the behavior of Confluent Schema Registry when
+transcoding collection documents into Avro for use with Dekaf,
+which must encode each message with an Avro schema ID (registry_id).
+';
+
+alter table registered_avro_schemas enable row level security;
+
+create policy "Users must be read-authorized to the schema catalog name"
+  on registered_avro_schemas as permissive
+  using (exists(
+    select 1 from auth_roles('read') r where catalog_name ^@ r.role_prefix
+  ));
+
+grant select on registered_avro_schemas to authenticated;
+grant insert (catalog_name, avro_schema) on registered_avro_schemas to authenticated;
+grant update (updated_at) on registered_avro_schemas to authenticated;
```

Comment:
This seems weird. Why would we want users to update only the `updated_at` column? Seems like it should include the schema, too?

Reply:
Registered schemas need to be immutable -- once a registry ID is encoded on the wire, data has been coded to the schema and the recipient needs to be able to retrieve the exact same schema in order to reliably decode it.

However, it would be really nice to have a measure of "which of these schemas aren't actually used anymore?" so that we could expunge them / reclaim portions of the registry_id space, and that was the role I figured `updated_at` could play.